toolchain/check/testdata/generic/dot_self_symbolic_type.carbon

```diff
@@ -0,0 +1,738 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// EXTRA-ARGS: --dump-sem-ir-ranges=if-present
```

Comment:
Can you document why you're using `if-present` instead of `only` here?

Reply:
Is this just about the lookup_impl_witness bit below?

Reply:
Yeah. That instruction is somehow ending up with a location that points to another `InstId`, which is in no block and has no location, which results in the `lookup_impl_witness` instruction not appearing in the formatted output if any ranges are present.

I've been digging into why it ends up with no location, but I've not figured it out yet.

Reply:
Figured it out, fixed, ranges added.

Reply:
I'm glad you fixed this. FWIW, I was considering we could also unconditionally emit insts in function bodies with no location, although maybe we could instead just CHECK-fail in that case?

Reply:
(to be sure, just mentioning in general -- I might look into options later)